### ISSUE TITLE
Replimat cant feed newborns (alpha animals/genetics expanded) 

### ISSUE THREAD

**Issue by Onslaught96** [sumghai/Replimat#74]
**Describe the bug:**  

colonists will not request baby food from terminal to feed babies
**Expected behavior:**  

colonist to request and receive baby food and proceed to feed baby
**Steps to reproduce:**  

1. build replimat terminal, feedstock, computer via debugmode
2. fill feedstock tank via debugmode
3. spawn newborn, adopt
4. try to feed newborn with colonist
**Additional information:**

Root level exception in OnGUI(): System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.FoodUtility.WillEat_NewTemp (Verse.Pawn p, Verse.ThingDef food, Verse.Pawn getter, System.Boolean careIfNotAcceptableForTitle, System.Boolean allowVenerated) [0x0000d] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at Replimat.ReplimatUtility.RepMatWillEat (Verse.Pawn p, Verse.ThingDef food, Verse.Pawn getter) [0x00008] in <deff2b9011c94e99921eb083a9570a5c>:0 
  at Replimat.ReplimatUtility+<>c__DisplayClass8_0.<PickMeal>b__0 (Verse.ThingDef x) [0x00000] in <deff2b9011c94e99921eb083a9570a5c>:0 
  at System.Linq.Enumerable+WhereEnumerableIterator`1[TSource].ToList () [0x0001b] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at Replimat.ReplimatUtility.PickMeal (Verse.Pawn eater, Verse.Pawn getter) [0x00061] in <deff2b9011c94e99921eb083a9570a5c>:0 
  at Replimat.Harmony_FoodUtility_GetFinalIngestibleDef.Prefix (Verse.Thing& foodSource, Verse.ThingDef& __result) [0x00010] in <deff2b9011c94e99921eb083a9570a5c>:0 
  at (wrapper dynamic-method) RimWorld.FoodUtility.RimWorld.FoodUtility.GetFinalIngestibleDef_Patch1(Verse.Thing,bool)
  at (wrapper dynamic-method) RimWorld.FoodUtility.RimWorld.FoodUtility.SpawnedFoodSearchInnerScan_Patch1(Verse.Pawn,Verse.IntVec3,System.Collections.Generic.List`1<Verse.Thing>,Verse.AI.PathEndMode,Verse.TraverseParms,single,System.Predicate`1<Verse.Thing>)
  at (wrapper dynamic-method) RimWorld.FoodUtility.RimWorld.FoodUtility.BestFoodSourceOnMap_NewTemp_Patch2(Verse.Pawn,Verse.Pawn,bool,Verse.ThingDef&,RimWorld.FoodPreferability,bool,bool,bool,bool,bool,bool,bool,bool,bool,bool,bool,RimWorld.FoodPreferability,System.Nullable`1<single>,bool)
  at RimWorld.FoodUtility.TryFindBestFoodSourceFor_NewTemp (Verse.Pawn getter, Verse.Pawn eater, System.Boolean desperate, Verse.Thing& foodSource, Verse.ThingDef& foodDef, System.Boolean canRefillDispenser, System.Boolean canUseInventory, System.Boolean canUsePackAnimalInventory, System.Boolean allowForbidden, System.Boolean allowCorpse, System.Boolean allowSociallyImproper, System.Boolean allowHarvest, System.Boolean forceScanWholeMap, System.Boolean ignoreReservations, System.Boolean calculateWantedStackCount, System.Boolean allowVenerated, RimWorld.FoodPreferability minPrefOverride) [0x00172] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.ChildcareUtility.FindBabyFoodForBaby (Verse.Pawn feeder, Verse.Pawn baby) [0x00012] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.WorkGiver_BottleFeedBaby.JobOnThing (Verse.Pawn pawn, Verse.Thing t, System.Boolean forced) [0x00014] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.WorkGiver_Scanner.HasJobOnThing (Verse.Pawn pawn, Verse.Thing t, System.Boolean forced) [0x00000] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.FloatMenuMakerMap.AddJobGiverWorkOrders (UnityEngine.Vector3 clickPos, Verse.Pawn pawn, System.Collections.Generic.List`1[T] opts, System.Boolean drafted) [0x001a9] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.FloatMenuMakerMap.AddUndraftedOrders (UnityEngine.Vector3 clickPos, Verse.Pawn pawn, System.Collections.Generic.List`1[T] opts) [0x00085] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at (wrapper dynamic-method) RimWorld.FloatMenuMakerMap.RimWorld.FloatMenuMakerMap.ChoicesAtFor_Patch1(UnityEngine.Vector3,Verse.Pawn,bool)
  at RimWorld.FloatMenuMakerMap.TryMakeFloatMenu (Verse.Pawn pawn) [0x000dc] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.Selector.HandleMapClicks () [0x000ac] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.Selector.SelectorOnGUI () [0x00000] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.MapInterface.HandleLowPriorityInput () [0x0000f] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at RimWorld.UIRoot_Play.UIRootOnGUI () [0x000dd] in <a3ecb7f3699146939cf9b29bca0b2137>:0 
  at (wrapper dynamic-method) Verse.Root.Verse.Root.OnGUI_Patch1(Verse.Root)
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
(wrapper dynamic-method) Verse.Root:Verse.Root.OnGUI_Patch1 (Verse.Root)
## System and Game Configuration
**Operating System:** Microsoft Windows 10
**Physical RAM:** 8.00 GB
**RimWorld base game version:** 1.4.3558
**RimWorld DLCs loaded:** Royalty, Ideology, Biotech
**Replimat mod version:** 1.4.3
**Replimat mod downloaded from:** Steam
**Your mod list:** 
Harmony
Hugslib 10.0.1
Replimat v1.4.3
Vanilla Expanded Framework
Alpha Animals
Vanilla Genetics Expanded
Vanilla Genetics Expanded - More Lab Stuff
## Declaration

I hereby verify that:
- [x] I am running the latest version of the game, Replimat and its dependencies
- [x] I am NOT running any mods with known incompatibilities with Replimat 
- [x] I have narrowed down my modlist to the fewest mods needed to cause this issue
- [x] I have confirmed that the issue does not happen if Replimat is disabled
- [x] I have searched the issue tracker and confirmed no-one else has reported my issue before
- [x] I understand that my issue will be closed and ignored if I do not fill this form properly

**Comment by Onslaught96** [comment]
Not entirely sure if bug needs to be resolved by VE but did notice similar update from 21st May without the additional context of newborns

Version 1.3.2 - 21 May 2022

Updated for 1.3.3326
Updated compatibility patches and functions
Fixed an issue where the Vanilla Cooking Expanded patch was referencing content from Vanilla Plants Expanded
Fixed NullReferenceException error preventing pawns from using Replimat Terminals when Alpha Animals or Vanilla Genetics Expanded mods are active

**Comment by sumghai** [comment]
Right, I've reproduced the issue with just Biotech, Replimat and **_either_** Alpha Animals **_or_** Vanilla Genetics Expanded (with or without More Lab Stuff).

Possibly related to #73, where Vanilla Animals Expanded - Caves introduced a food item without an `ingestible` XML node, and the issue was ultimately fixed on Sarg's end.

Let me investigate further and see if this is the case.

**Comment by Onslaught96** [comment]
Alright. thanks for the quick reply!

**Comment by sumghai** [comment]
Just as I suspected, Sarg had a whole bunch of fertilized egg-like items from all his mods that were missing the `ingestible` node, which broke the base game's `RimWorld.FoodUtility.WillEat_NewTemp()`

Although I was able to convince him to fix VAE - Caves back in #73, there are so many items from Alpha Animals and Vanilla Genetics Expanded with the same type of bug, that he may be reluctant to apply the same fixes (even if I submit pull requests for all his mods).

Best thing to do would be to tell Replimat's `RepMatWillEat()` to skip execution early for any food that is missing the `ingestible` node, and return `false`. This will act as a safeguard against other badly-coded mods.
